File: google/cloud/securitycenter/settings/v1beta1/google-cloud-securitycenter-settings-v1beta1-ruby/lib/google/cloud/security_center/settings/v1beta1/doc/google/cloud/securitycenter/settings/v1beta1/billing_settings.rb
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

module Google
  module Cloud
    module Securitycenter
      module Settings
        module V1beta1
          # Billing settings
          # @!attribute [rw] billing_tier
          #   @return [Google::Cloud::Securitycenter::Settings::V1beta1::BillingTier]
          #     Output only. Billing tier selected by customer
          # @!attribute [rw] billing_type
          #   @return [Google::Cloud::Securitycenter::Settings::V1beta1::BillingType]
          #     Output only. Type of billing method
          # @!attribute [rw] start_time
          #   @return [Google::Protobuf::Timestamp]
          #     Output only. The absolute point in time when the subscription became effective.
          #     Can be compared to expire_time value to determine full contract duration
          # @!attribute [rw] expire_time
          #   @return [Google::Protobuf::Timestamp]
          #     Output only. The absolute point in time when the subscription expires.
          #
          #     If this field is populated and billing_tier is STANDARD, this is
          #     indication of a point in the _past_ when a PREMIUM access ended.
          class BillingSettings; end

          # Billing tier options
          module BillingTier
            # Default value. This value is unused.
            BILLING_TIER_UNSPECIFIED = 0

            # The standard billing tier.
            STANDARD = 1

            # The premium billing tier.
            PREMIUM = 2
          end

          # Billing type
          module BillingType
            # Default billing type
            BILLING_TYPE_UNSPECIFIED = 0

            # Subscription for Premium billing tier
            SUBSCRIPTION = 1

            # Trial subscription for Premium billing tier
            TRIAL_SUBSCRIPTION = 2

            # Alpha customer for Premium billing tier
            ALPHA = 3
          end
        end
      end
    end
  end
end

File: google/cloud/securitycenter/settings/v1beta1/securitycenter-settings-v1beta1-py/google/cloud/securitycenter/settings_v1beta1/types/billing_settings.py
# -*- coding: utf-8 -*-

# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

import proto  # type: ignore


from google.protobuf import timestamp_pb2 as timestamp  # type: ignore


__protobuf__ = proto.module(
    package='google.cloud.securitycenter.settings.v1beta1',
    manifest={
        'BillingTier',
        'BillingType',
        'BillingSettings',
    },
)


class BillingTier(proto.Enum):
    r"""Billing tier options"""
    BILLING_TIER_UNSPECIFIED = 0
    STANDARD = 1
    PREMIUM = 2


class BillingType(proto.Enum):
    r"""Billing type"""
    BILLING_TYPE_UNSPECIFIED = 0
    SUBSCRIPTION = 1
    TRIAL_SUBSCRIPTION = 2
    ALPHA = 3


class BillingSettings(proto.Message):
    r"""Billing settings

    Attributes:
        billing_tier (google.cloud.securitycenter.settings_v1beta1.types.BillingTier):
            Output only. Billing tier selected by
            customer
        billing_type (google.cloud.securitycenter.settings_v1beta1.types.BillingType):
            Output only. Type of billing method
        start_time (google.protobuf.timestamp_pb2.Timestamp):
            Output only. The absolute point in time when the
            subscription became effective. Can be compared to
            expire_time value to determine full contract duration
        expire_time (google.protobuf.timestamp_pb2.Timestamp):
            Output only. The absolute point in time when the
            subscription expires.

            If this field is populated and billing_tier is STANDARD,
            this is indication of a point in the *past* when a PREMIUM
            access ended.
    """

    billing_tier = proto.Field(proto.ENUM, number=1,
        enum='BillingTier',
    )

    billing_type = proto.Field(proto.ENUM, number=2,
        enum='BillingType',
    )

    start_time = proto.Field(proto.MESSAGE, number=3,
        message=timestamp.Timestamp,
    )

    expire_time = proto.Field(proto.MESSAGE, number=4,
        message=timestamp.Timestamp,
    )


__all__ = tuple(sorted(__protobuf__.manifest))
